sgt-back-end/server.ts

```diff
@@ -0,0 +1,130 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError } from './client-error.js';
+import { errorMiddleware } from './error-middleware.js';
+
+// only create ONE pool for your whole server
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+      select *
+        from "grades"
+    `;
+    const result = await db.query(sql);
+    res.json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const gradeId = Number(req.params.gradeId);
+    console.log(gradeId);
+    if (!Number.isInteger(gradeId) || gradeId <= 0) {
+      throw new ClientError(400, '"gradeId" must be a positive integer');
+    }
+    const sql = `
+      select *
+        from "grades"
+        where "gradeId" = $1
+    `;
+    const params = [gradeId];
+    const result = await db.query(sql, params);
+    const grade = result.rows[0];
+    if (!grade) {
+      throw new ClientError(404, `Cannot find grade with "gradeId" ${gradeId}`);
+    }
+    res.json(grade);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/grades', async (req, res, next) => {
+  try {
+    const { name, course, score } = req.body;
+    const sql = `
+      insert into "grades" ("name", "course", "score")
+        values ($1, $2, $3)
+        returning *;
+    `;
+    const params = [name, course, score];
+    const result = await db.query(sql, params);
+    const grade = result.rows;
+    if (!name || !course || !score) {
```

Comment:
You should do this check _before_ issuing the SQL. If the input data is bad, you don't want to put it in the database -- and the database may throw an exception which will result in a 500 error, so the client will never learn that they sent you bad data.